test/gtest/ucp/test_ucp_am.cc

```diff
@@ -1906,3 +1913,88 @@ UCS_TEST_P(test_ucp_am_nbx_rndv_memtype_disable_zcopy,
 }
 
 UCP_INSTANTIATE_TEST_CASE_GPU_AWARE(test_ucp_am_nbx_rndv_memtype_disable_zcopy);
+
+
+#ifdef ENABLE_STATS
+class test_ucp_am_nbx_rndv_ppln : public test_ucp_am_nbx_rndv {
+public:
+    void init() override
+    {
+        stats_activate();
+        modify_config("RNDV_THRESH", "128");
+        test_ucp_am_nbx::init();
+    }
+
+    void cleanup() override
+    {
+        test_ucp_am_nbx::cleanup();
+        stats_restore();
+    }
+
+    static void get_test_variants(variant_vec_t &variants)
+    {
+        if (!mem_buffer::is_gpu_supported()) {
+            return;
+        }
+
+        add_variant_with_value(variants, UCP_FEATURE_AM, 0, "");
+    }
+
+protected:
+    void test_ppl_send(ucs_memory_type_t mem_type, size_t num_frags,
+                       uint64_t sender_cntr, uint64_t sender_cntr_value,
+                       uint64_t receiver_cntr, uint64_t receiver_cntr_value)
+    {
+        if (!has_rma_zcopy()) {
+            UCS_TEST_SKIP_R("No RMA support");
+        }
+
+        const size_t rndv_frag_size = get_rndv_frag_size(mem_type);
+        test_am_send_recv(rndv_frag_size * num_frags);
+
+        check_stats(sender(), sender_cntr, sender_cntr_value);
+        check_stats(receiver(), receiver_cntr, receiver_cntr_value);
+    }
+
+private:
+    void check_stats(entity &e, uint64_t cntr, uint64_t exp_value)
+    {
+        auto stats_node = e.worker()->stats;
+        auto value      = UCS_STATS_GET_COUNTER(stats_node, cntr);
+
+        EXPECT_EQ(exp_value, value) << "counter is "
+                                    << stats_node->cls->counter_names[cntr];
+    }
+
+    bool has_rma_zcopy() {
+        return has_transport("rc_v") || has_transport("rc_x") ||
+               has_transport("dc_x") ||
+               (ucp_context_find_tl_md(receiver().ucph(), "cma")  != nullptr) ||
+               (ucp_context_find_tl_md(receiver().ucph(), "knem") != nullptr);
+    }
+};
+
+UCS_TEST_SKIP_COND_P(test_ucp_am_nbx_rndv_ppln, host_buff_cuda_frag,
+                     !(is_proto_enabled()),
+                     "RNDV_SCHEME=put_ppln", "RNDV_FRAG_MEM_TYPE=cuda")
```

Comment:
does it use cuda_ipc?

Reply:
why do you mean? cuda_ipc can be used with pipeline protocols, but here we specify memory type for ppln bounce buffers